models/HubSpotHttpError.ts

```diff
@@ -14,6 +14,7 @@ export class HubSpotHttpError<T = any> extends Error {
   public headers?: { [key: string]: unknown };
   public method: string | undefined;
   public context: HubSpotHttpErrorContext | undefined;
+  public derivedContext: HubSpotHttpErrorContext | undefined;
```

Comment:
Splitting the context derived into it's own field so when the provided context is updated we don't slip details into the log the user didn't intend.